services/src/main/java/org/keycloak/services/util/CookieHelper.java

```diff
@@ -106,20 +108,15 @@ public static void addCookie(String name, String value, String path, String doma
 
     private static Set<String> getInternalCookieValue(String name) {
         HttpHeaders headers = Resteasy.getContextData(HttpHeaders.class);
-
         Set<String> cookiesVal = new HashSet<>();
 
         // check for cookies in the request headers
-        List<String> cookieHeader = headers.getRequestHeaders().get(HttpHeaders.COOKIE);
-        if (cookieHeader != null) {
-            logger.debugv("{1} cookie found in the request's header", name);
-            cookieHeader.stream().map(s -> parseCookie(s, name)).forEach(cookiesVal::addAll);
-        }
+        cookiesVal.addAll(parseCookie(headers.getRequestHeaders().getFirst(HttpHeaders.COOKIE), name));
```

Comment:
It seems that with current KEycloak we support parsing more "Cookie" headers and even same cookies in each header. Something like this:
```
Cookie: foo=val1
Cookie: foo=val2
```
With this change, we will always have just the "val1" from the first cookie if I understand correctly? I don't have clue about side-effects of this and if there are some regressions of this and some (old) browsers will stop working as expected etc. Not sure if this is same what @martin-kanis mentioned?

Reply:
I suggest to add some test to "CookieHelper" as well for the format like this. To doublecheck that we don't have regressions. WDYT?

Reply:
As per https://tools.ietf.org/html/rfc6265#section-5.4, "When the user agent generates an HTTP request, the user agent MUST NOT attach more than one Cookie header field.".

As per parsing multiple cookies within the same header, the test is checking that. As long as they have different values.

But again, I'm not sure if this code is necessary for two main reasons:

* Old and EOL browser support seems to be the reason behind that code
* Resteasy should parse the cookies just like that already ?

Plus the fact that we are adding some overhead on every request to parse the header whenever a cookie is queried.

Reply:
@mposolda It's still working with multiple cookies scenario which you described. It should be covered by the following test `CookiesPathTest.testMultipleCookies`. 

@pedroigor I agree that Resteasy parser for parsing cookies from a header is definitely better than the previous custom one. It's now really a question whether we want to support a wrong behavior of IE11 and live with this performance penalty. Another argument why keep it might be related to product as I assume there are more strict rules regarding the IE11 support.

Reply:
We can live with it, mainly due to the fact that otherwise we won't be able to parse multiple values for a same cookie. And that is one of the things this changeset tries to fix where the same cookie is sent multiple times but due to the wrong parsing no cookies are actually parsed.